test/brain/testing_index_selection_util.cpp

```diff
@@ -0,0 +1,335 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// testing_index_selection_util.cpp
+//
+// Identification: test/brain/testing_index_selection_util.cpp
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "brain/testing_index_selection_util.h"
+#include "brain/what_if_index.h"
+#include "common/harness.h"
+#include "optimizer/stats/stats_storage.h"
+#include "sql/testing_sql_util.h"
+#include "planner/index_scan_plan.h"
+
+namespace peloton {
+
+namespace test {
+
+namespace index_selection {
+
+TestingIndexSelectionUtil::TestingIndexSelectionUtil(std::string db_name)
+    : database_name_(db_name) {
+  srand(time(NULL));
+  CreateDatabase();
+}
+
+TestingIndexSelectionUtil::~TestingIndexSelectionUtil() {
+  for (auto it = tables_created_.begin(); it != tables_created_.end(); it++) {
+    DropTable(it->first);
+  }
+  DropDatabase();
+}
+
+std::pair<std::vector<TableSchema>, std::vector<std::string>>
+TestingIndexSelectionUtil::GetQueryStringsWorkload(
```

Comment:
I think this is an awesome utility given that theres a lot of self-driving work that can use something like this now(like ours :)) and in the future. 
A few suggestions here would be:
1. Adding a metric utility here - which given the query and hypothetical configuration can provide the "what-if" cost directly. I know we can call your internal classes and do ourselves - but exposing an API would just be so much easier for everyone. Other metrics can then be added over time like the one we discussed today(number of tuples read.)
2. Is there any specific reason these specific workloads have been chosen - does each one have some specific characteristic - it might be worthwhile to appropriately name and add comments about these workloads.

Reply:
1. When we designed the utility, we thought of one that can be used for testing the What-If API as well. But if everyone needs it such an interface we will add it later.

2. These are some simple workloads which can expose all the (logical) branches in the index selection algorithm.